### PR TITLE
Improve logic to handle new available networks

### PR DESCRIPTION
**Pull request by alalamav**
* When a VPN is established, `connectivityManager.getActiveNetworkInfo()`, returns the underlying network set by the VPN. 
* Use the network passed to `onAvailable` to set the underlying network instead of the active network, and do not perform a comparison between them.
* This change properly sets the VPN's underlying network for Android API 23+, meaning that apps that require WiFi connectivity now work correctly with the VPN.
* Tested that network changes are still handled on Android API <23.

### CODE REVIEW

File: cordova-plugin-outline/android/java/org/outline/vpn/VpnTunnelService.java [hunk 1]
```diff
@@ -353,10 +353,7 @@ public void onAvailable(Network network) {
       NetworkInfo activeNetworkInfo = connectivityManager.getActiveNetworkInfo();
       LOG.fine(String.format(Locale.ROOT, "Network available: %s\nActive network: %s", networkInfo,
```
Comment (bemasc) [inline cordova-plugin-outline/android/java/org/outline/vpn/VpnTunnelService.java]: This log message now seems misleading

**Comment by alalamav** [comment]
I have verified that when WiFi is on (without cellular), turning on cellular does not trigger an `onAvailable` event. Turning off WiFi while having cellular connectivity does trigger it.

I also verified that `connectivityManager.getActiveNetwork()` returns the underlying network set by the VPN.

File: cordova-plugin-outline/android/java/org/outline/vpn/VpnTunnelService.java (hunk 1 above)
Comment (alalamav) [reply cordova-plugin-outline/android/java/org/outline/vpn/VpnTunnelService.java]: Fixed in 480ccfb.

**Comment by bemasc** [comment]
I'm a bit concerned about relying on this undocumented behavior of `onAvailable`.  What if `onAvailable` is called for all new `Network`s on some versions, and only for "preferable" networks on your test version?  The documentation doesn't clearly indicate the latter behavior.

**Pull request merged**